### PR TITLE
Updating spraycans and crayons.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Porting in TGstation PRs #40559, #42327, #43146, #43358, #44122 and #44561 (also some apparently direct commits by Naksu that made some spraycan/crayons lists into static/lists, unsure if it's a better choice than globals in this context).

## Why It's Good For The Game
Updating the code, adding in the possibility of using spraycans to color object, new crayon decals and making its tgui less laggy.

## Changelog
:cl: Ghommie (original PRs by Mickyan, Anturk, ShizCalev, nemvar and Naksu)
add: After rigorous mandatory art training for the crew, many new graffiti styles are now available
fix: Cleaned up some crayon and spraycan code for futureproofing.
fix: Spraypainting blast doors no longer makes them see-through.
balance: Paint remover now works on blast doors and the like.
add: Most objects can now be colored using a spray can.
spellcheck: Added visible message to spraying objects and windows.
add: Colored lights now shine in different colours.
del: Removed individual buttons text in crayon/spraycan UI, speeding it up.
fix: Text mode buffer is actually visible in the UI.
tweak: Last letter of a text mode buffer no longer rotates out to be replaced with "a", allowing the text mode to be used for individual symbols.
/:cl:

### CODE REVIEW

File: code/game/objects/items/crayons.dm [hunk 1]
```diff
@@ -32,16 +35,16 @@
 	var/drawtype
 	var/text_buffer = ""
 
-	var/list/graffiti = list("amyjon","face","matt","revolution","engie","guy","end","dwarf","uboa","body","cyka","arrow","star","poseur tag","prolizard","antilizard", "tile") //cit edit
-	var/list/letters = list("a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z")
-	var/list/numerals = list("0","1","2","3","4","5","6","7","8","9")
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/crayons.dm:37]: Did you mean to remove arrow, letters and numerals?

Comment (Ghommie) [reply code/game/objects/items/crayons.dm:37]: Yes, letters and numbers considerably slown down the UI load a lot.
People can still write things down with the text buffer anyway, so the functionality is still there.
About the `arrow` entry, it's already present on the `oriented` list.

**Review by silicons — DISMISSED** [review]

**Comment by nik707** [comment]
@Ghommie fix conflicts and it's merged

**Comment by Ghommie** [comment]
conflicts solved and has been tested locally successfully @nik707.

**Review by silicons — APPROVED** [review]

**Pull request merged**